zebra-state/src/service/finalized_state/zebra_db/shielded.rs

```diff
@@ -355,6 +436,92 @@ impl DiskWriteBatch {
             self.zs_insert(&orchard_tree_cf, height, trees.orchard);
         }
 
+        if let Some(subtree) = trees.sapling_subtree {
+            let (subtree_end, subtree_start) = subtree.bounds();
+
+            let start_height = if subtree_end.index == 0 {
+                let (start_height, _) = zebra_db
+                    .sapling_tree_by_height_range(Height(1)..)
+                    .next()
+                    .expect("must have sapling notes on disk");
+
+                start_height
+            } else if let Some(start_height) =
+                zebra_db.sapling_subtree_start_height_by_index(subtree_end.index)
+            {
+                start_height
+            } else {
+                let prev_subtree_end_height = zebra_db
+                    .sapling_subtree_by_index(subtree_end.index - 1)
+                    .expect("must have previous subtree")
+                    .end;
+
+                let (start_height, _) = zebra_db
+                    .sapling_tree_by_height_range(prev_subtree_end_height.next()..)
+                    .next()
+                    .expect("must have previous subtree");
+
+                start_height
+            };
+
+            self.zs_insert(
+                &sapling_subtree_cf,
+                subtree_end.index,
+                (start_height, subtree_end.height, subtree_end.root),
+            );
+
+            if let Some(subtree_start) = subtree_start {
+                self.zs_insert(
+                    &sapling_subtree_cf,
+                    subtree_start.index,
+                    subtree_start.height,
+                );
+            }
```

Comment:
Is it OK to write two different data types into the same column family?

Reply:
I think both keys and values are an array of bytes, so it should be okay.

Reply:
> Currently the serialization will panic if the wrong type is read:

Oh. I'll fix that in the `FromDisk` impl.

> How does zcashd do it?

I'm not sure zcashd is tracking the start heights on main or any open PRs.

Reply:
I think this code has now been removed.